pallets/handles/src/lib.rs

```diff
@@ -0,0 +1,628 @@
+		///
+		/// # Returns
+		///
+		/// * `PresumptiveSuffixesResponse` - The response containing the next available suffixes.
+		/// ```
+		pub fn get_next_suffixes(input: PresumptiveSuffixesRequest) -> PresumptiveSuffixesResponse {
+			let handle = input.base_handle;
+			let count = input.count;
+			let mut suffixes: Vec<HandleSuffix> = vec![];
+			let base_handle: Handle = handle.try_into().unwrap_or_default();
+			let base_handle_str = core::str::from_utf8(&base_handle).unwrap_or("");
+
+			let canonical_handle_str = HandleConverter::convert_to_canonical(&base_handle_str);
+			let canonical_handle_vec = canonical_handle_str.as_bytes().to_vec();
+			let canonical_handle: Handle = canonical_handle_vec.try_into().unwrap();
+			let suffix_index =
+				Self::get_next_suffix_index_for_canonical_handle(canonical_handle.clone())
+					.unwrap_or_default();
+
+			// Generate suffixes from the next available suffix index
+			let mut suffix_generator = SuffixGenerator::new(
+				T::HandleSuffixMin::get() as usize,
+				T::HandleSuffixMax::get() as usize,
+				&canonical_handle_str,
+			);
+			let sequence = suffix_generator.suffix_iter().enumerate();
+
+			for (i, suffix) in sequence {
+				if i >= suffix_index as usize && i < (suffix_index as usize + count as usize) {
+					suffixes.push(suffix as u16);
+				}
+				if suffixes.len() == count as usize {
+					break
+				}
+			}
+			let response =
+				PresumptiveSuffixesResponse { base_handle: base_handle.into(), suffixes };
+			response
+		}
+
+		/// Creates a full display handle by combining a base handle string with a suffix generated
+		/// from an index into the suffix sequence.
+		///
+		/// # Arguments
+		///
+		/// * `base_handle_str` - The base handle string.
+		/// * `suffix_sequence_index` - The index into the suffix sequence.
+		///
+		/// # Returns
+		///
+		/// * `Handle` - The full display handle.
+		///
+		#[cfg(test)]
+		pub fn create_full_handle_for_index(
+			base_handle_str: &str,
+			suffix_sequence_index: SequenceIndex,
+		) -> Vec<u8> {
+			// Convert base display handle into a canonical display handle
+			let canonical_handle_str = HandleConverter::convert_to_canonical(&base_handle_str);
+
+			// Generate suffix from index into the suffix sequence
+			let suffix = Self::generate_suffix_for_canonical_handle(
+				&canonical_handle_str,
+				suffix_sequence_index as usize,
+			);
+
+			let full_handle = Self::create_full_handle(base_handle_str, suffix);
+			full_handle
+		}
+
+		/// Creates a full display handle by combining a base handle string with supplied suffix
+		///
+		/// # Arguments
+		///
+		/// * `base_handle_str` - The base handle string.
+		/// * `suffix` - The numeric suffix .
+		///
+		/// # Returns
+		///
+		/// * `Handle` - The full display handle.
+		///
+		#[cfg(test)]
+		pub fn create_full_handle(base_handle_str: &str, suffix: HandleSuffix) -> Vec<u8> {
```

Comment:
I think we can remove this [test] and make this a helper function. This code is a subset of do_claim_handle.

Reply:
Tagged as a todo